src/modules/visitor-space/components/AutocompleteFieldInput/AutocompleteFieldInput.tsx

```diff
@@ -0,0 +1,77 @@
+import { type SelectOption } from '@meemoo/react-components';
+import clsx from 'clsx';
+import { type FC, useCallback } from 'react';
+import { type ActionMeta, type SingleValue } from 'react-select';
+import AsyncCreatableSelect from 'react-select/async-creatable';
+
+import { IeObjectsService } from '@ie-objects/services';
+import { tText } from '@shared/helpers/translate';
+
+import styles from './AutocompleteFieldInput.module.scss';
+
+export interface AutocompleteFieldInputProps {
+	label?: string;
+	fieldName: string;
+	disabled?: boolean;
+	id?: string;
+	onChange: (value: string | null) => void;
+	value?: string;
+	className?: string;
+}
+
+const AutocompleteFieldInput: FC<AutocompleteFieldInputProps> = ({
+	onChange,
+	value,
+	fieldName,
+	label,
+}) => {
+	const handleLoadOptions = useCallback(
+		(inputValue: string, callback: (newOptions: SelectOption[]) => void): void => {
+			if (inputValue.length < 3) {
+				callback([]);
+				return;
+			}
+			IeObjectsService.getAutocompleteFieldOptions(fieldName, inputValue).then((options) => {
+				callback(options.map((option) => ({ label: option, value: option })));
+			});
+		},
+		[fieldName]
+	);
+
+	const handleChange = (
+		newValue: SingleValue<SelectOption>,
+		actionMeta: ActionMeta<SelectOption>
+	): void => {
+		if (actionMeta.action === 'select-option' || actionMeta.action === 'create-option') {
+			onChange(newValue?.value || null);
+		}
+	};
```

Comment:
Is there a need for this different handleChange component. Can't this be don direcly from the `<AsyncCreatableSelect<SelectOption>` component

Reply:
you mean: inline in the html code?
i think its cleaner to keep the html code as minimal as possible and make any onChange that isn't just calling the onChange of the props a separate handler in the code.